restapi/src/main/java/io/stargate/web/docsapi/service/ReactiveDocumentService.java

```diff
@@ -111,15 +113,30 @@ public ReactiveDocumentService(
   }
 
   private Single<Integer> determineTtl(
-      Boolean ttlAuto, DocumentDB db, String namespace, String collection, String documentId) {
-    return Single.fromCallable(
-        () -> {
-          if (ttlAuto == null || !ttlAuto) {
-            return 0;
-          }
-          authorizeRead(db, namespace, collection);
-          return db.getTtlForDocument(namespace, collection, documentId);
-        });
+      Boolean ttlAuto, DocumentDB db, String namespace, String collection, String documentId)
+      throws UnauthorizedException {
+    if (ttlAuto == null || !ttlAuto) {
+      return Single.just(0);
+    }
```

Comment:
can we remove the ttlAuto here completely please? this method should be called only when there is a need to get a TTL from a document, clean and easy..